docs/consensus/registry.md

```diff
@@ -0,0 +1,155 @@
+[`SignedEntity`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/common/entity?tab=doc#SignedEntity
+[`Entity`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/common/entity?tab=doc#Entity
+[signed envelope]: ../crypto.md#signed-envelope
+[escrow account]: staking.md#escrow
+<!-- markdownlint-enable line-length -->
+
+### Deregister Entity
+
+Entity deregistration enables an existing entity to be removed. A new deregister
+entity transaction can be generated using [`NewDeregisterEntityTx`].
+
+**Method name:**
+
+```
+registry.DeregisterEntity
+```
+
+The body of a register entity transaction must be `nil`. The entity is implied
+to be the signer of the transaction.
+
+_If an entity still has either nodes or runtimes registered, it is not possible
+to deregister an entity and such a transaction will fail._
+
+<!-- markdownlint-disable line-length -->
+[`NewDeregisterEntityTx`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/registry/api?tab=doc#NewDeregisterEntityTx
+<!-- markdownlint-enable line-length -->
+
+### Register Node
+
+Node registration enables a new node to be created. A new register node
+transaction can be generated using [`NewRegisterNodeTx`].
+
+**Method name:**
+
+```
+registry.RegisterNode
+```
+
+The body of a register entity transaction must be a [`MultiSignedNode`]
+structure, which is a [multi-signed envelope] containing a [`Node`] descriptor.
+The signer of the transaction MUST be the node identity key.
+
+The owning entity MUST have the given node identity public key whitelisted in
+the `Nodes` field in its [`Entity`] descriptor.
+
+The node descriptor structure MUST be signed by all of the following keys:
+
+* Node identity key.
+* Consensus key.
+* Committee TLS certificate key.
+* Committee P2P key.
+
+Registering a node may require sufficient stake in the owning entity's
+[escrow account].
+
+<!-- markdownlint-disable line-length -->
+[`NewRegisterNodeTx`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/registry/api?tab=doc#NewRegisterNodeTx
+[`MultiSignedNode`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/common/node?tab=doc#MultiSignedNode
+[`Node`]: https://pkg.go.dev/github.com/oasislabs/oasis-core/go/common/node?tab=doc#Node
+[multi-signed envelope]: ../crypto.md#multi-signed-envelope
+<!-- markdownlint-enable line-length -->
+
+### Unfreeze Node
+
+Node unfreezing enables a previously frozen (e.g., due to slashing) node to be
+thawed so it can again be eligible for committee elections. A new unfreeze node
+transaction can be generated using [`NewUnfreezeNodeTx`].
+
+**Method name:**
+
+```
+registry.UnfreezeNode
+```
+
+**Body:**
+
+```golang
+type UnfreezeNode struct {
+    NodeID signature.PublicKey `json:"node_id"`
+}
+```
+
+**Fields:**
+
+* `node_id` specifies the node identifier of the node to thaw.
+
+The transaction signer MUST be the entity key that owns the node.
+
+Thawing a node requires that the node's freeze period has already passed.
```

Comment:
Might be useful to note that the duration of the freeze period for a given event (e.g., double signing) is a consensus parameter.